source/Timer.h

```diff
@@ -0,0 +1,110 @@
+/* Timer.h
+Copyright (c) 2023 by Timothy Collett
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef TIMER_H_
+#define TIMER_H_
+
+#include "MissionAction.h"
+#include "System.h"
+
+#include <cstdint>
+#include <map>
+#include <string>
+
+class DataNode;
+class DataWriter;
+class Planet;
+class PlayerInfo;
+class UI;
+
+// Class representing a timer for triggering mission actions
+// This requires a specification of how long to wait (base + randint(rand)),
+// a system to wait in, and whether to reset the timer upon leaving the system
+// and can optionally require the player to be idle (no acceleration), and/or
+// within or beyond a certain proximity to the system center or a
+// specific (named) system object
+class Timer {
+public:
+	Timer() = default;
+	Timer(const DataNode &node, Mission *mission);
+	// Set up the timer from its data file node
+	void Load(const DataNode &node, Mission *mission);
```

Comment:
You could take the mission by `const` here

Reply:
OK; that does make sense, as it's only there to pass to `GameAction::Do()`. Thanks.